pkgs/development/libraries/at-spi2-atk/default.nix

```diff
@@ -1,24 +1,47 @@
-{ stdenv, fetchurl, python, pkgconfig, popt, atk, libX11, libICE, xorg, libXi
-, intltool, dbus-glib, at-spi2-core, libSM }:
+{ stdenv
+, fetchurl
+
+, removeReferencesTo
+, meson
+, ninja
+, pkgconfig
+
+, at-spi2-core
+, atk
+, dbus
+, glib
+, libxml2
+
+, gnome3 # Only to pass update script
+}:
+
+with stdenv.lib;
 
 stdenv.mkDerivation rec {
-  versionMajor = "2.26";
-  versionMinor = "1";
-  moduleName   = "at-spi2-atk";
-  name = "${moduleName}-${versionMajor}.${versionMinor}";
+  name = "${pName}-${version}";
+  pName = "at-spi2-atk";
+  version = "2.26.2";
 
   src = fetchurl {
-    url = "mirror://gnome/sources/${moduleName}/${versionMajor}/${name}.tar.xz";
-    sha256 = "0x9vc99ni46fg5dzlx67vbw0zqffr24gz8jvbdxbmzyvc5xw5w5l";
+    url = "mirror://gnome/sources/${pName}/${versions.majorMinor version}/${name}.tar.xz";
+    sha256 = "0vkan52ab9vrkknnv8y4f1cspk8x7xd10qx92xk9ys71p851z2b1";
   };
 
-  nativeBuildInputs = [ pkgconfig intltool ];
-  buildInputs = [ python popt atk libX11 libICE xorg.libXtst libXi
-                  dbus-glib at-spi2-core libSM ];
+  nativeBuildInputs = [ removeReferencesTo meson ninja pkgconfig ];
```

Comment:
What does `removeReferencesTo` do?

Reply:
I picked it up from Docker package.
It is new and rarely user.
It deletes packages in array with it from closure.

It used to delete build-only time dependencies from closure.
So for example, when package is pulled from cache, it does not pull `meson` `ninja` and etc. that are in the same array with them.

Reply:
Well, those packages should not get into closure. As mentioned in https://github.com/NixOS/nixpkgs/pull/41365#issuecomment-394786610, the build would have to include their store paths to the resulting package. If that happened, it should be reported as a bug. Just run `nix-store -q --references result/` to check the closure.

Also just adding `removeReferencesTo` to dependencies will not do anything – you would have to call `remove-references-to` manually in one of the phases.